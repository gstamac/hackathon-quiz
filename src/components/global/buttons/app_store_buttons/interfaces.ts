export interface AppstoresInterface {
  text: string
  href: string
  iconLink: string
}
