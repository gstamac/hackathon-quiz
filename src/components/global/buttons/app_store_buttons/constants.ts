import { APPLE_STORE_LINK, GOOGLE_STORE_LINK } from '../../../../constants'
import androidAppstoreIconLink from '../../../../assets/icons/android_appstore_btn.png'
import iosAppstoreIconLink from '../../../../assets/icons/ios_appstore_btn.png'
import { AppstoresInterface } from './interfaces'

export const appStores: AppstoresInterface[] = [
  {
    text: 'IOS appstore',
    href: APPLE_STORE_LINK,
    iconLink: iosAppstoreIconLink,
  },
  {
    text: 'Android appstore',
    href: GOOGLE_STORE_LINK,
    iconLink: androidAppstoreIconLink,
  },
]
