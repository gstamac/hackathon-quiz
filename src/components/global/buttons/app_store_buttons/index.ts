export { AppStoreButtons } from './app_store_buttons'
