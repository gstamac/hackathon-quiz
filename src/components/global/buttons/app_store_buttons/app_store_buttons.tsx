import React from 'react'
import { useStyles } from '../../dialogs/styled_dialog/style'
import { appStores } from './constants'
import { AppstoresInterface } from './interfaces'

export const AppStoreButtons: React.FC = () => {
  const { appStoreLinks } = useStyles()

  return (
    <div className={appStoreLinks} data-testid='app-store-links'>
      {appStores.map(({ text, href, iconLink }: AppstoresInterface, i: number) => (
        <a href={href} key={i}>
          <img src={iconLink} alt={text} />
        </a>
      ))}
    </div>
  )
}
