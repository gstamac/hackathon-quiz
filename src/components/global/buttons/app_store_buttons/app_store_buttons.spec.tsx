import React from 'react'
import { render, RenderResult } from '../../../../../tests/test_utils'
import { AppStoreButtons } from './app_store_buttons'

describe('AppStoreButtons', () => {
  it('should render app store links', () => {
    const renderResult: RenderResult = render(<AppStoreButtons />)

    expect(renderResult.queryAllByTestId('app-store-links')).not.toBeNull()
    expect(renderResult.container).toMatchSnapshot()
  })
})
