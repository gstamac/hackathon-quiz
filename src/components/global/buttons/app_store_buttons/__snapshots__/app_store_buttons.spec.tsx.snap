// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppStoreButtons should render app store links 1`] = `
<div>
  <div
    class="makeStyles-appStoreLinks-1"
    data-testid="app-store-links"
  >
    <a
      href="https://apps.apple.com/us/app/globalid-portable-identity/id1439340119"
    >
      <img
        alt="IOS appstore"
        src="ios_appstore_btn.png"
      />
    </a>
    <a
      href="https://play.google.com/store/apps/details?id=net.globalid"
    >
      <img
        alt="Android appstore"
        src="android_appstore_btn.png"
      />
    </a>
  </div>
</div>
`;
